src/server.rs

```diff
@@ -684,4 +690,45 @@ impl WalletService for Arc<crate::wallet::Wallet> {
             "not implemented",
         ))
     }
+
+    async fn ripemd160(
+        &self,
+        request: tonic::Request<Ripemd160Request>,
+    ) -> std::result::Result<tonic::Response<Ripemd160Response>, tonic::Status> {
+        let message = request.into_inner().message;
+        let message: Vec<u8> = message
+            .ok_or_else(|| missing_field::<GetBmmHStarCommitmentRequest>("message"))?
+            .decode_tonic::<GetBlockHeaderInfoRequest, _>("message")?;
+        let mut hasher = ripemd::Ripemd160::new();
+        hasher.update(&message);
+        let hash = hasher.finalize();
+        let response = Ripemd160Response {
+            hash: Some(ConsensusHex::encode(&hash.as_slice().to_vec())),
```

Comment:
Hashes are fixed size, so you should convert to an array, not a vec/slice, which will encode the length prefix.
use `hash.to_byte_array()`

Reply:
There is no `to_byte_array()` in `GenericArray`:

https://docs.rs/generic-array/0.14.7/generic_array/struct.GenericArray.html

Reply:
see https://github.com/LayerTwo-Labs/bip300301_enforcer/pull/63/files#r1815525170, this should be done via the `bitcoin` crate